android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt

```diff
@@ -273,22 +275,23 @@ class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaM
 
         val trackList = Arguments.toList(data)
         val queue = musicService.tracks
-        val indexes: MutableList<Int> = ArrayList()
-        for (o in trackList!!) {
-            val index = if (o is Int) o else o.toString().toInt()
-
-            // We do not allow removal of the current item
-            musicService.getCurrentTrackIndex {
-                if (index == it) return@getCurrentTrackIndex
-                if (index >= 0 && index < queue.size) {
-                    indexes.add(index)
+
+        if (trackList != null) {
+            for (track in trackList) {
+                val index = if (track is Int) track else track.toString().toInt()
+
+                // We do not allow removal of the current item
+                musicService.getCurrentTrackIndex {
+                    if (index == it) {
+                        Timber.d("This track is currently playing, so it can't be removed")
+                        return@getCurrentTrackIndex
+                    } else if (index >= 0 && index < queue.size) {
+                        musicService.remove(listOf(index))
+                    }
                 }
             }
         }
 
-        if (indexes.isNotEmpty())
-            musicService.remove(indexes)
```

Comment:
We are trying to remove indexes before `getCurrentTrackIndex` has finished collecting all the indexes.